packages/vite-plugin-cloudflare/src/runner-worker/module-runner.ts

```diff
@@ -54,7 +55,12 @@ export async function createModuleRunner(
 			async runInlinedModule(context, transformed, module) {
 				if (
 					module.file.includes("/node_modules") &&
-					!module.file.includes("/node_modules/.vite")
+					!module.file.includes("/node_modules/.vite") &&
+					// if the code includes some of the vite context functions (e.g. `__vite_ssr_exports__`)
+					// we assume that the module was processed by vite, meaning that even if it is in the
+					// node_modules folder it is actually not an external module (this can happen for node_module assets
+					// imported via vite's assets API: https://vite.dev/guide/assets.html#static-asset-handling)
+					!Object.keys(context).some((k) => transformed.includes(k))
```

Comment:
This is clearly not super robust (unlikely but some library could include strings such as `__vite_ssr_exports__` in their source code 🤷)

alternative we could check the `module.id` to see if it ends with `?url`, `?raw`, etc...

I would be ok with that solution if people would prefer it (I think I slightly prefer the above one but barely), the pro there is that it would be a quite robust solution, the con is that it would be more hardcoded and slightly less maintainable (we'd have to list the various vite asset import types here and keep them in sync with vite's)

Reply:
Trying to remember why this if statement is here. Was is the scenario where this error would correctly be reported?

Reply:
I agree both options look meh. With my experience I would suggest avoiding this one:
> alternative we could check the module.id to see if it ends with ?url, ?raw, etc...

Plugins can extend the list of asset import types and users can often customise it which would likely mean a continuous stream of the same sort of issue. Take the [vite-plugin-svgr](https://www.npmjs.com/package/vite-plugin-svgr) for example which adds a `?react` which can be customised as you wish:


Maybe an alternative could be to look for the `?` in general? - The question mark character is disallowed in the `package.json` - `name` field. So if there is a `?` char then some "other" processing is definitely happening.

Reply:
I agree. The `?` on its own should be a signal that Vite is processing the import in some way.

Reply:
Vite also adds `?v=` on non-processed imports though

are you suggesting of checking for the presence of `?` and also checking if the `v` search param is present?

e.g.:
![Screenshot 2025-03-26 at 11 54 26](https://github.com/user-attachments/assets/2019cfc6-9834-4b5a-b85e-e898a130a3d3)

Reply:
I've removed these validations in favour of config validations as discussed below